packages/eslint-plugin-eds/src/rules/no-p-tags.ts

```diff
@@ -0,0 +1,21 @@
+import type {Rule} from 'eslint';
+
+const failureMessage =
+  'Please use the <Text> component from the @chanzuckerberg/eds repo instead of a <p> tag 🥰';
+
+const rule: Rule.RuleModule = {
+  create(context) {
+    return {
+      JSXElement(node: any) {
+        if (node.openingElement.name.name === 'p') {
+          context.report({
```

Comment:
@jinlee93 I was thinking we could add an extra check here on `node.parent` -- could be a different API but iirc you can access the parent (and grandparent etc. nodes) from the AST that way. It's the same AST-walking syntax as codemod files so could put the example in an AST explorer and check what the structure would be.

- [rough example from jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/93f78856655696a55309440593e0948c6fb96134/src/rules/anchor-has-content.js#L40)
- [example of accessing a grandparent node](https://blog.maximeheckel.com/posts/how-to-build-first-eslint-rule/#create)

Reply:
To resolve this thread, paraphrasing from other thread:  the `as="div"` / textpassage variant of the `Text` component is deprecated so there’s no need to add additional functionality to check if the parent / grandparent is `Text`